### PR TITLE
#505 Add more fields definition for maven in packagedcode

### PR DESCRIPTION
**Pull request by linexb**
Add more POM supported field to get more rich data from pom file

### CODE REVIEW

**Comment by pombredanne** [comment]
@linexb Thanks! Let me review this

File: src/packagedcode/maven.py [hunk 1]
```diff
@@ -92,37 +92,49 @@ def recognize(cls, location):
     ('component_parent_version', '/project/parent/version'),
     ('component_packaging', '/project/packaging'),
     ('component_version', '/project/version'),
-
+
```
Comment (pombredanne) [inline src/packagedcode/maven.py]: You have a few extra spaces there.

File: tests/packagedcode/data/maven/expectations/activemq-camel-pom.xml.json [hunk 2]
```diff
@@ -31,10 +34,25 @@
   "maven_component_group_id": [
     "org.apache.activemq"
   ], 
+  "maven_contributor_email": [], 
   "maven_distribution_management_group_id": [], 
   "maven_scm_developer_connection": [], 
   "maven_component_extend_1": [], 
   "maven_developer_name": [], 
+  "maven_contributor_organization": [], 
+  "maven_dependency": [
```
Comment (pombredanne) [inline tests/packagedcode/data/maven/expectations/activemq-camel-pom.xml.json]: I think the approach we have with XPATH in the current maven code is flawed (not fault from you but from me). We are missing the actual groupid, the scope, optional, etc for each dep. This needs to be a proper data structure and not a plain aligned lists....

**Comment by pombredanne** [comment]
@linexb Note that I rebased your branch on the latest develop

File: src/packagedcode/maven.py (hunk 1 above)
Comment (linexb) [inline src/packagedcode/maven.py]: I have removed the extra spaces.

**Comment by pombredanne** [comment]
@linexb I squashed and reworded the commits but I am not sure I grok exactly where we are going with this. For sure this maven module is a mess (that I created) and we need to clean it up. Piling up up more features on top can only look weirder.

**Comment by pombredanne** [comment]
This has been replaced by #561and #529 which are merged in develop. So I am closing this without merging.

**Pull request closed without merging**